CMakeLists.txt

```diff
@@ -261,6 +252,22 @@ set(MIOPEN_hip_VERSION ${MIOPEN_hip_VERSION_MAJOR}.${MIOPEN_hip_VERSION_MINOR}.$
 set_var_to_condition(MIOPEN_USE_HIPRTC_DEFAULT (MIOPEN_USE_COMGR AND (MIOPEN_hip_VERSION VERSION_GREATER_EQUAL 5)))
 option(MIOPEN_USE_HIPRTC "Use HIPRTC to build HIP kernels instead of COMGR" ${MIOPEN_USE_HIPRTC_DEFAULT})
 
+# Do not append system include directories to HIP compiler flags when HIPRTC is used
+set_var_to_condition(MIOPEN_HIP_COMPILER_USE_SYSTEM_INCLUDE_DIRECTORIES_DEFAULT
+              (NOT (MIOPEN_USE_HIPRTC AND (MIOPEN_hip_VERSION VERSION_GREATER_EQUAL 6.1.40091))))
```

Comment:
`6.1.40091` seems questionable. IIRC the patch version format is yywwd, which means that this is 6.1 built at Sunday week 09 year 2040. As a result, system includes will be always used until 6.2 (please check am I correct or not)

If we want to disable system includes starting from 6.1, let's use "6.1.0".

Reply:
we are relying on hip_version rather than rocm release version.

The hip_version.h file  from 6.1.0 RC2  build shows the version as below. So hip version corresponding to 6.1.0 is 6.1.40091
#define HIP_VERSION_MAJOR 6
#define HIP_VERSION_MINOR 1
#define HIP_VERSION_PATCH 40091

MIOPEN_hip_VERSION is populated as below. 
set(MIOPEN_hip_VERSION ${MIOPEN_hip_VERSION_MAJOR}.${MIOPEN_hip_VERSION_MINOR}.${MIOPEN_hip_VERSION_PATCH})

In [6.0.2 RC1 GA] the corresponding version is 6.0.24015

Reply:
@raramakr yes, please check, maybe they changed the format due to some reason. 

[resolved]

Reply:
Sure @atamazov and thank you for reviewing the patch
If any changes in hip_VERSION , will upload a separate patch to update the same

Reply:
@raramakr 🔴 hip_VERSION should maintain backward compatibility. The goal is that any new version of MIOpen works with some previous version of HIP correctly. If HIP guys change the format, and MIOpen adapts to it, then MIOpen may loose the capability to with with older HIP. So the question is important.

**_Can you please check with HIP team if they changed the format or not._** So far, I am assuming that `40091` is just a glitch.

/cc @junliume

Reply:
sure @atamazov

Reply:
@atamazov @junliume, From hip team
The format of the 5 digit is :- YdddX.
Y = last digit of the year. So 4 for 2024. 9 for 2029.
ddd = day in year. So goes 001 to 366.
X = is initially set to 0 on every mainline promotion. Increments by 1 if we ever make a cherry-pick to the point release.

In 6.0.2 the patch version is 32831, but cmake is reporting hip_VERSION as 6.0.24015 (24015- HIP_VERSION_GITDATE)
In 6.1.0 the patch version is 40091 and cmake is reporting hip_VERSION  as  6.1.40091. (40091- HIP_VERSION_PATCH)
There was change in 6.1 for HIP patch version reported via CMake , and that is why the difference in behavior
-    VERSION "${HIP_VERSION_MAJOR}.${HIP_VERSION_MINOR}.${**HIP_VERSION_GITDATE**}"
+    VERSION "${HIP_VERSION_MAJOR}.${HIP_VERSION_MINOR}.${**HIP_VERSION_PATCH**}"

Reply:
@raramakr [sarcasm to HIP guys] So what we will do on the 2029/2030 tripping point? Let's have another kind of Y2K problem each 10 years instead of 100 years, right? 🤣

Reply:
@raramakr Seriously, thanks for finding this out and informing us 🤝

Reply:
[resolved] But please do not hide this review comment.

Reply:
Since the comparison is between versions(Major.Minor.patch) of each releases,  the patch version should not be an issue, since major and minor will change with each release

Reply:
> @raramakr [sarcasm to HIP guys] So what we will do on the 2029/2030 tripping point? Let's have another kind of Y2K problem each 10 years instead of 100 years, right? 🤣

Thanks @raramakr for sharing! This has puzzled me for a long time :) After year of 2029 the major version would be very different then.